src/code/qualification/plotQualificationPKRatio.m

```diff
@@ -51,23 +71,33 @@
     SimResult = loadSimResultcsv(csvSimFile, PKRatio.Simulation);
     
     % All the output are kept so far, may be removed if not necessary
-    [AGE, BW, MW, drugmass] = getInfofromSimulation(xmlfile, PKRatio.Output);
+    [AGE, BW, MW, drugmass, drugmassUnit] = getInfofromSimulation(xmlfile, PKRatio.Output);
     
     Result.AGE(i, 1)=AGE;
     Result.BW(i, 1)=BW;
     Result.MW(i, 1)=MW;
     Result.drugmass(i, 1)=drugmass(1);
+    Result.drugmassUnit{i, 1}=drugmassUnit(1);
     
     % Get the right PK Output
+    if isempty(SimResult.outputPathList)
+        ME = MException('plotQualificationPKRatio:emptyOutputPathInSimulation', ...
+            'In PK Ratio plot %d, Ratio %d, OutputPath is empty in Project "%s" or Simulation "%s"', figureHandle, i, PKRatio.Project, PKRatio.Simulation);
+        throw(ME);
+    end
     for j=1:length(SimResult.outputPathList)
         findPathOutput = strfind(SimResult.outputPathList{j}, PKRatio.Output);
         if ~isempty(findPathOutput)
-            Pred=SimResult.y{j};
-            PredUnit=SimResult.outputUnit{j};
-            SimTime=SimResult.time;
             SimTimeUnit=SimResult.timeUnit;
+            PredUnit=SimResult.outputUnit{j};
+            TimeUnitFactor=getUnitFactor(SimTimeUnit, 'h', 'Time');
+            PredUnitFactor=getUnitFactor(PredUnit, 'mg/l', 'Concentration', 'MW', MW);
+            
+            % Get Time in h and Concentraiton in g/l

```

Comment:
First question: why do we convert to 'mg/l' and to 'h' here?
Second question: comment states 'g/l' (either the comment or the conversion is wrong)

Reply:
1) I wanted to have a known unique unit for flow, AUC and Cmax. I don't think I can know the unit after using getPKParametersForConcentration, so I converted everything before using it.
I think based on PKSim standard unit, concentration may be always in umol/ml, time in min and dose in umol. If it is the case, I can assume the standard unit to convert into the observed unit.
2) I started in g/l, wasn't in the unit list, so went for mg/l but forgot to update. I'm going to commit that change.

Reply:
Internal PK-Sim units are:
* `"µmol/L"` for **Concentration**
* `"min"` for **Time**

And times/concentration in simulation results CSVs are always in internal units (@msevestre correct me if I'm wrong)

Thus if times/concentrations in the internal units are passed to getPKParametersForConcentration, units for calculated PK-Parameters would be (@KatrinCoboeken correct me if I'm wrong)
* `"µmol/l"` for **CMax** (same as concentration)
* `"µmol*min/l"` for **AUC**
* `"l/min"` for **CL**

Reply:
Ok. I'll use directly these units then if that's okay.

Reply:
By the end it's only important to use the **same units** when calculating `PKParameter.Predicted/PKParameter.Observed`.
Thus either predicted PKParameter must be converted into the unit of observed PK Parameter or vice versa

Reply:
> 
> 
> Ok. I'll use directly these units then if that's okay.

should be ok